chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -896,6 +903,21 @@ func (n *NeutrinoNotifier) RegisterConfirmationsNtfn(txid *chainhash.Hash,
 	return ntfn.Event, nil
 }
 
+// GetBlock is used to retrieve the block with the given hash. Since the block
+// cache used by neutrino will be the same as that used by LND (since it is
+// passed to neutrino on initialisation), the neutrino GetBlock method can be
+// called directly since it already uses the block cache. However, neutrino
+// does not lock the block cache mutex for the given block hash and so that is
+// done here.
+func (n *NeutrinoNotifier) GetBlock(hash chainhash.Hash) (
+	*btcutil.Block, error) {
+
+	n.blockCache.LockHash(hash)
```

Comment:
Looks like we need to do this for the neutrino implementation of `lnwallet.BlockChainIO` as well?

Reply:
ah yes 👍  good catch. Will update 👍

Reply:
added in 0eb1f42c7ac95ef3b3b15d560416232f3941c1eb 👍